Exception thrown when no arguments are passed to "add student" command

### Description
When no arguments are passed to the "add student" command, a `NullPointerException` is thrown.

### Steps to reproduce
1. Open the app.
2. Type in the command "add student" and press "Enter".

### Expected behaviour
Error message of invalid command format displayed.

### Actual behaviour
Nothing displayed as an exception is thrown.

```
Caused by: java.lang.NullPointerException
Caused by: java.lang.NullPointerException

	at seedu.address.logic.parser.ArgumentTokenizer.findPrefixPosition(ArgumentTokenizer.java:73)
	at seedu.address.logic.parser.ArgumentTokenizer.findPrefixPositions(ArgumentTokenizer.java:50)
	at seedu.address.logic.parser.ArgumentTokenizer.lambda$findAllPrefixPositions$0(ArgumentTokenizer.java:40)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at seedu.address.logic.parser.ArgumentTokenizer.findAllPrefixPositions(ArgumentTokenizer.java:41)
	at seedu.address.logic.parser.ArgumentTokenizer.tokenize(ArgumentTokenizer.java:27)
	at seedu.address.logic.parser.AddCommandParser.parse(AddCommandParser.java:37)
	at seedu.address.logic.parser.AddressBookParser.parseCommand(AddressBookParser.java:59)
	at seedu.address.logic.LogicManager.execute(LogicManager.java:45)
	at seedu.address.ui.MainWindow.executeCommand(MainWindow.java:177)
	at seedu.address.ui.CommandBox.handleCommandEntered(CommandBox.java:45)
	... 59 more
```

--- a/src/main/java/seedu/address/logic/parser/AddressBookParser.java
+++ b/src/main/java/seedu/address/logic/parser/AddressBookParser.java
@@ -35,7 +35,6 @@ public class AddressBookParser {
      * @throws ParseException if the user input does not conform the expected format
      */
     public Command parseCommand(String userInput) throws ParseException {
-        boolean isTwoWordCommand = false;
         final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
         if (!matcher.matches()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
@@ -43,16 +42,14 @@ public class AddressBookParser {
 
         String commandWord = matcher.group("commandWord");
         String arguments = matcher.group("arguments");
-        if ((arguments.length() != 0) && (!arguments.startsWith(" -"))) {
-            isTwoWordCommand = true;
-            if (Character.isDigit(arguments.charAt(1))) {
-                isTwoWordCommand = false;
-            }
-        }
+        boolean isTwoWordCommand = arguments.length() > 0
+                && !arguments.startsWith(" -") && !Character.isDigit(arguments.charAt(1));
+
         if (isTwoWordCommand) {
-            commandWord = convertFormat(commandWord, arguments);
-            arguments = convertFormatArguments(arguments);
+            commandWord = extractFullCommandWord(commandWord, arguments);
+            arguments = extractArguments(arguments);
         }
+
         switch (commandWord) {
 
         case AddCommand.COMMAND_WORD:
@@ -87,40 +84,32 @@ public class AddressBookParser {
     /**
      * Removes second word of command from arguments.
      *
-     * @param arguments arguments to format.
-     * @return formatted arguments.
+     * @param arguments raw arguments to extract from.
+     * @return extracted arguments.
      */
-    private String convertFormatArguments(String arguments) {
-        int index = 0;
-        if (arguments.contains("-")) {
-            index = arguments.indexOf("-");
+    private String extractArguments(String arguments) {
+        int argumentsIndex = arguments.indexOf("-");
+        if (argumentsIndex == -1) {
+            return "";
         }
-        if (index == 0) {
-            return null;
-        } else {
-            String argumentsFormatted = arguments.substring(index - 1);
-            return argumentsFormatted;
-        }
+        String extractedArguments = arguments.substring(argumentsIndex - 1);
+        return extractedArguments;
     }
 
     /**
      * Converts commandWord to the two word format and retrieves second word from arguments.
      *
-     * @param commandWord commandWord to format.
+     * @param firstCommandWord the original commandWord parsed.
      * @param arguments arguments to retrieve second half of commandWord from.
-     * @return formatted commandWord.
+     * @return full commandWord.
      */
-    private String convertFormat(String commandWord, String arguments) {
-        int index = 0;
-        if (arguments.contains("-")) {
-            index = arguments.indexOf("-");
+    private String extractFullCommandWord(String firstCommandWord, String arguments) {
+        int argumentsIndex = arguments.indexOf("-");
+        if (argumentsIndex == -1) {
+            return firstCommandWord + arguments;
         }
-        if (index == 0) {
-            return commandWord + arguments;
-        } else {
-            String commandWordPartTwo = commandWord + arguments.substring(0, index - 1);
-            return commandWordPartTwo;
-        }
+        String fullCommandWord = firstCommandWord + arguments.substring(0, argumentsIndex - 1).stripTrailing();
+        return fullCommandWord;
     }
 
 }

